app/helpers/otp_helper.rb

```diff
@@ -0,0 +1,53 @@
+module OtpHelper
+  def phone_confirmation_instructions
+    if @delivery_method == 'sms'
+      t('instructions.2fa.confirm_code_sms', number: @phone_number)
+    elsif @delivery_method == 'voice'
+      t('instructions.2fa.confirm_code_voice', number: @phone_number)
+    end
+  end
+
+  def fallback_2fa_links
+    case @delivery_method
+    when 'sms'
+      "#{voice_fallback_link}#{totp_option_link}."
+    when 'voice'
+      "#{sms_fallback_link}#{totp_option_link}."
+    when 'recovery-code'
+      t('devise.two_factor_authentication.recovery_code_help', phone: phone_fallback_link)
+    else
+      ''
+    end
+  end
+
+  def authenticator_link
+    link_to(t('devise.two_factor_authentication.totp_name'), login_two_factor_authenticator_path)
+  end
+
+  def recovery_code_fallback_link
+    link_to(t('devise.two_factor_authentication.recovery_code_fallback.link'),
+            login_two_factor_recovery_code_path)
+  end
+
+  private
```

Comment:
no private methods in helpers http://stackoverflow.com/questions/12835367/why-can-private-helper-methods-still-be-accessed-in-views :( 

Another reason it might be good to move this stuff into presenter classes!